src/cljs/rems/actions/change_resources.cljs

```diff
@@ -0,0 +1,121 @@
+(ns rems.actions.change-resources
+  (:require [clojure.string :as str]
+            [re-frame.core :as rf]
+            [rems.actions.action :refer [action-button action-form-view action-comment button-wrapper collapse-action-form]]
+            [rems.autocomplete :as autocomplete]
+            [rems.catalogue-util :refer [get-catalogue-item-title]]
+            [rems.common-util :refer [index-by]]
+            [rems.spinner :as spinner]
+            [rems.status-modal :as status-modal]
+            [rems.text :refer [text]]
+            [rems.util :refer [fetch post!]]))
+
+(rf/reg-event-fx
+ ::open-form
+ (fn
+   [{:keys [db]} [_ initial-resources]]
+   (merge
+    {:db (assoc db
+                ::comment ""
+                ::initial-resources (into #{} (map :catalogue-item/id initial-resources))
+                ::selected-resources (into #{} (map :catalogue-item/id initial-resources)))}
+    (when-not (:rems.catalogue/catalogue db)
+      {:dispatch [:rems.catalogue/fetch-catalogue]}))))
+
+(rf/reg-event-db ::set-sorting (fn [db [_ sorting]] (assoc db ::sorting sorting)))
+(rf/reg-sub ::sorting (fn [db _] (::sorting db)))
+
+(rf/reg-event-db ::set-filtering (fn [db [_ filtering]] (assoc db ::filtering filtering)))
+(rf/reg-sub ::filtering (fn [db _] (::filtering db)))
+
+(defn resource-matches? [language resource query]
+  (-> (get-catalogue-item-title resource language)
+      .toLowerCase
+      (.indexOf query)
+      (not= -1)))
+
+(rf/reg-sub ::initial-resources (fn [db _] (::initial-resources db)))
+(rf/reg-sub ::selected-resources (fn [db _] (::selected-resources db)))
+(rf/reg-event-db ::set-selected-resources (fn [db [_ resources]] (assoc db ::selected-resources resources)))
+(rf/reg-event-db ::add-selected-resources (fn [db [_ resource]] (update db ::selected-resources conj (:id resource))))
+(rf/reg-event-db ::remove-selected-resource (fn [db [_ resource]] (update db ::selected-resources disj (:id resource))))
+
+(rf/reg-sub ::comment (fn [db _] (::comment db)))
+(rf/reg-event-db ::set-comment (fn [db [_ value]] (assoc db ::comment value)))
```

Comment:
Sub usually after effect handler

Reply:
I think they are pretty random in our code.